src/main/java/com/example/contactsservice/Service/ContactService.java

```diff
@@ -0,0 +1,68 @@
+package com.example.contactsservice.Service;
+
+import com.example.contactsservice.BadRequest;
+import com.example.contactsservice.DTO.ContactDTO;
+import com.example.contactsservice.Entity.ContactEntity;
+import com.example.contactsservice.NotFound;
+import com.example.contactsservice.Repository.ContactRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ContactService {
+    @Autowired
+    ContactRepository contactRepository;
+
+    public ContactEntity createContact(ContactDTO contactDTO) {
+        validateContactDto(contactDTO);
+        ContactEntity contact = getContact(contactDTO, new ContactEntity());
+        return contactRepository.save(contact);
+    }
+
+    public ContactEntity getContact(Long id) {
+        ContactEntity contact = contactRepository.getById(id);
+        try {
+            if (null != contact.getFirstName()) {
+                return contact;
+            }
+            return null;
+        } catch (RuntimeException runtimeException) {
+            throw new NotFound("User doesn't exist");
+        }
+    }
+
+    public ContactEntity updateContact(ContactDTO contactDTO, Long id) {
+        validateContactDto(contactDTO);
+        ContactEntity contact = getContact(id);
+        contact.setFirstName(contactDTO.getFirstName());
+        contact.setLastName(contactDTO.getLastName());
+        contact.setPhone(contactDTO.getPhone());
+        return contactRepository.save(contact);
+    }
+
+    public void deleteContact(Long id) {
+        ContactEntity contact = getContact(id);
+        if (null != contact) {
+            contactRepository.delete(contact);
+        } else {
+            throw new NotFound("User doesn't exist");
```

Comment:
what if you get the error during deletion of the contact? you have not handled that error here.